collect_app/src/main/java/org/odk/collect/android/widgets/utilities/RangeWidgetUtils.java

```diff
@@ -0,0 +1,118 @@
+import org.odk.collect.android.utilities.ToastUtils;
+import org.odk.collect.android.views.CustomRangeSlider;
+
+import java.math.BigDecimal;
+
+public class RangeWidgetUtils {
+    private static final String VERTICAL_APPEARANCE = "vertical";
+
+    private RangeWidgetUtils() {
+    }
+
+    public static Object[] setUpLayoutElements(Context context, FormEntryPrompt prompt) {
+        View answerView;
+        CustomRangeSlider slider;
+        TextView currentValue;
+        TextView minValue;
+        TextView maxValue;
+
+        String appearance = prompt.getQuestion().getAppearanceAttr();
+
+        if (appearance != null && appearance.contains(VERTICAL_APPEARANCE)) {
+            RangeWidgetVerticalBinding rangeWidgetVerticalBinding = RangeWidgetVerticalBinding
+                    .inflate(((Activity) context).getLayoutInflater());
+            answerView = rangeWidgetVerticalBinding.getRoot();
+
+            slider = rangeWidgetVerticalBinding.slider;
+            currentValue = rangeWidgetVerticalBinding.currentValue;
+            minValue = rangeWidgetVerticalBinding.minValue;
+            maxValue = rangeWidgetVerticalBinding.maxValue;
+        } else {
+            RangeWidgetHorizontalBinding rangeWidgetHorizontalBinding = RangeWidgetHorizontalBinding
+                    .inflate(((Activity) context).getLayoutInflater());
+            answerView = rangeWidgetHorizontalBinding.getRoot();
+
+            slider = rangeWidgetHorizontalBinding.slider;
+            currentValue = rangeWidgetHorizontalBinding.currentValue;
+            minValue = rangeWidgetHorizontalBinding.minValue;
+            maxValue = rangeWidgetHorizontalBinding.maxValue;
+        }
+        setUpWidgetParameters((RangeQuestion) prompt.getQuestion(), minValue, maxValue);
+
+        if (prompt.isReadOnly()) {
+            slider.setEnabled(false);
+        }
+
+        return new Object[] {answerView, slider, currentValue};
+    }
+
+    static void setUpWidgetParameters(RangeQuestion rangeQuestion, TextView minValue, TextView maxValue) {
+        minValue.setText(String.valueOf(rangeQuestion.getRangeStart()));
+        maxValue.setText(String.valueOf(rangeQuestion.getRangeEnd()));
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    public static void setUpSlider(RangeQuestion rangeQuestion, CustomRangeSlider slider, BigDecimal actualValue) {
+        BigDecimal rangeStart = rangeQuestion.getRangeStart();
+        BigDecimal rangeEnd = rangeQuestion.getRangeEnd();
+        BigDecimal rangeStep = rangeQuestion.getRangeStep().abs();
+
+        slider.setValueFrom(rangeStart.floatValue());
+        slider.setValueTo(rangeEnd.floatValue());
+        slider.setStepSize(rangeStep.floatValue() != 0 ? rangeStep.floatValue() : 0.5F);
+        slider.setValue(actualValue == null ? rangeStart.floatValue() : actualValue.floatValue());
+
+        slider.setOnTouchListener((v, event) -> {
+            int action = event.getAction();
+            switch (action) {
```

Comment:
What is this code for?

Reply:
@seadowg I think this just adds a touch listener on the `slider`. Without it, I was not able to set the value on the vertical slider. `requestDisallowInterceptTouchEvent()` does not allow the parent of the child to intercept the touch events acoording to the [docs](https://developer.android.com/reference/android/view/ViewParent). It disallows the parent `ViewGroup` to intercept swipe up and swipe down the action and lets the user set the values on the slider.

Reply:
Hmmm. Yeah tricky. Could we maybe make this part of the custom `Slider` implementation? It's probably very awkward to test but at least if it's in there (and with a comment) it's more obviously important!